src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -410,7 +415,11 @@ void ServiceBasedActorInfoAccessor::AsyncResubscribe(bool is_pubsub_server_resta
     for (auto &item : subscribe_operations_) {
       auto &actor_id = item.first;
       RAY_CHECK_OK(item.second([this, actor_id](const Status &status) {
-        fetch_data_operations_[actor_id](nullptr);
+        absl::MutexLock lock(&mutex_);
+        auto fetch_data_operation = fetch_data_operations_[actor_id];
+        if (fetch_data_operation != nullptr) {
```

Comment:
Can you comment when this can be nullptr? (I guess unsubscribe and resubscribe are called at the same time by different threads?)

Reply:
fixed